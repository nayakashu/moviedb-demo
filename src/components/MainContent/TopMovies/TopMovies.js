import React, { Component } from 'react';
import { connect } from 'react-redux';
import MovieTile from '../MovieTile/MovieTile';
import Loader from '../../common/Loader/Loader';
import { closeLeftNav } from '../../../actions/navActions';
import './TopMovies.scss';

export class TopMovies extends Component {
  componentDidMount() {
    this.props.closeLeftNav();
  }

  render() {
    return (
      <div className="top-rated-movies">
        <h3>Top Rated Movies</h3>
        {this.props.topMovies ? null : <Loader />}
        <div className="row">
          {this.props.topMovies &&
            this.props.topMovies.map(item => (
              <div className="col-sm-4 col-xs-6" key={item.id}>
                <MovieTile movieDetails={item} />
              </div>
            ))}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  topMovies: state.movieListState.topMovies
});

export default connect(
  mapStateToProps,
  { closeLeftNav }
)(TopMovies);
