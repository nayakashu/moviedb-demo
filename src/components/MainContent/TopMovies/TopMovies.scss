@import '../../../Common.scss';

.top-rated-movies {
  h3 {
    margin-top: 30px;
    color: $text-color-light;
  }
}
