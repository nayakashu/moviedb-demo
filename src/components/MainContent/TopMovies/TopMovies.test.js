import React from 'react';
import { shallow } from 'enzyme';
import { TopMovies } from './TopMovies';

describe('TopMovies', () => {
  const topMovies = shallow(<TopMovies />);

  it('renders correctly', () => {
    expect(topMovies).toMatchSnapshot();
  });
});
