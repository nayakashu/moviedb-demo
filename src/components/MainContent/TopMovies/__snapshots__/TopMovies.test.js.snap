// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopMovies renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopMovies />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Top Rated Movies
        </h3>,
        <Loader />,
        <div
          className="row"
        />,
      ],
      "className": "top-rated-movies",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Top Rated Movies",
        },
        "ref": null,
        "rendered": "Top Rated Movies",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "row",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Top Rated Movies
          </h3>,
          <Loader />,
          <div
            className="row"
          />,
        ],
        "className": "top-rated-movies",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Top Rated Movies",
          },
          "ref": null,
          "rendered": "Top Rated Movies",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "row",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
